src/cargo/ops/cargo_add/mod.rs

```diff
@@ -599,19 +598,32 @@ fn select_package(
     }
 }
 
-fn infer_package(mut packages: Vec<Package>, src: &dyn std::fmt::Display) -> CargoResult<Package> {
+fn infer_package_for_git_source(
+    mut packages: Vec<Package>,
+    src: &dyn std::fmt::Display,
+) -> CargoResult<Package> {
     let package = match packages.len() {
-        0 => {
-            anyhow::bail!("no packages found at `{src}`");
-        }
+        0 => unreachable!(),
```

Comment:
Please add an explanation for why this is unreachable

Something like
```rust
unreachable!("this function should only be called with packages from `GitSource::read_packages` and that call should error for us when there are no packages");
```

Reply:
Good suggestion! Thanks!